File: app/src/main/java/com/example/tutor/login/ui/signup/SignupActivity.java
package com.example.tutor.login.ui.signup;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.tutor.MainActivity;
import com.example.tutor.R;
import com.google.firebase.auth.FirebaseAuth;

public class SignupActivity extends AppCompatActivity {
    private FirebaseAuth mAuth;
    EditText fullName, email, password, confirmPassword;
    Button signup;
    ImageView back;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_signup);

        fullName = findViewById(R.id.name);
        email = findViewById(R.id.email);
        password = findViewById(R.id.password);
        confirmPassword = findViewById(R.id.confirm_password);
        signup = findViewById(R.id.log_in);
        back = findViewById(R.id.back);

        back.setOnClickListener(v -> finish());

        mAuth = FirebaseAuth.getInstance();

        signup.setOnClickListener(v -> {
            String txt_email = email.getText().toString();
            String txt_password = password.getText().toString();

            if (TextUtils.isEmpty(txt_email) || TextUtils.isEmpty(txt_password)){
                Toast.makeText(
                        SignupActivity.this,
                        "Empty Credentials",
                        Toast.LENGTH_SHORT)
                        .show();
            }else if (txt_password.length() < 6){
                Toast.makeText(
                        SignupActivity.this,
                        "Password is too short",
                        Toast.LENGTH_SHORT)
                        .show();
            }else {
                registerUser(txt_email, txt_password);
                Intent intent = new Intent(SignupActivity.this, MainActivity.class);
                startActivity(intent);
                finish();
            }
        });
    }

    private void registerUser(String email, String password) {

        mAuth.createUserWithEmailAndPassword(email, password)
                .addOnCompleteListener(SignupActivity.this, task -> {
                    if (task.isSuccessful()){
                        Toast.makeText(
                                SignupActivity.this,
                                "User registration successful!",
                                Toast.LENGTH_SHORT)
                                .show();
                        startActivity(new Intent(SignupActivity.this, MainActivity.class));
                        finish();
                    }else {
                        Toast.makeText(
                                SignupActivity.this,
                                "Registration failed!", Toast.LENGTH_SHORT)
                                .show();
                    }
                });
    }
}

File: app/src/main/java/com/example/tutor/home/ui/signup/SignupActivity.kt
package com.example.tutor.home.ui.signup

import android.content.ContentValues.TAG
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.example.tutor.MainActivity
import com.example.tutor.R
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.auth.ktx.auth
import com.google.firebase.ktx.Firebase

class SignupActivity : AppCompatActivity() {
    // variables for the ids of the views
    private lateinit var fullname: EditText
    private lateinit var email: EditText
    private lateinit var password: EditText
    private lateinit var confirmPassword: EditText
    private lateinit var signupButton: Button
    private lateinit var auth: FirebaseAuth
    private lateinit var emailString: String
    private lateinit var passwordString: String
    private lateinit var confirmPasswordString: String
    private lateinit var fullnameString: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_signup)

        val toolbar: Toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)

        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowHomeEnabled(true)
        supportActionBar!!.setIcon(R.drawable.ic_arrow_back)

        auth = Firebase.auth

        // initializing the views
        fullname = findViewById(R.id.name)
        email = findViewById(R.id.email)
        password = findViewById(R.id.password)
        confirmPassword = findViewById(R.id.confirm_password)
        signupButton = findViewById(R.id.signup)

        signupButton.setOnClickListener {
            emailString = email.text.toString()
            passwordString = password.text.toString()
            confirmPasswordString = confirmPassword.text.toString()
            fullnameString = fullname.text.toString()
            if (passwordString == confirmPasswordString) {
                auth.createUserWithEmailAndPassword(emailString, passwordString)
                    .addOnCompleteListener(this) { task ->
                        if (task.isSuccessful) {
                            // Sign in success, update UI with the signed-in user's information
                            Log.w(TAG, "createUserWithEmail:success", task.exception)
                            Toast.makeText(
                                baseContext, "Authentication successful.",
                                Toast.LENGTH_SHORT
                            ).show()
                            val user = auth.currentUser
                            updateUI(user)
                            val intent = Intent(this@SignupActivity, MainActivity::class.java)
                            startActivity(intent)
                        } else {
                            // If sign in fails, display a message to the user.
                            Log.w(TAG, "createUserWithEmail:failure", task.exception)
                            Toast.makeText(
                                baseContext, "Authentication failed.",
                                Toast.LENGTH_SHORT
                            ).show()
                            updateUI(null)
                        }
                    }
            }
        }
    }

    private fun updateUI(user: FirebaseUser?) {
        // updater user UI
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}
